app/models/post.rb

```diff
@@ -23,8 +23,13 @@ def with_ordered
 
   scope :published, -> { where("published_at <= ?", Time.current) }
   scope :order_by_recent, -> { order(published_at: :desc, id: :asc) }
+  scope :categorized_by, lambda {|category|
+    category_ids = category.has_children? ? category.subtree_ids : category.id
```

Comment:
If this variable name is `category_ids`, can you always assign an instance of `Array`?

``` diff
- category.id
+ [category.id]
```